### PR TITLE
Phpstan

### LINKED ISSUE #34: Add PHPStan
This would be a good addition at one point.

### PR DESCRIPTION
**Pull request by ElectricMaxxx**
fix #34

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -49,6 +49,12 @@ matrix:
           # Dev-master is allowed to fail.
         - env: STABILITY="dev"
 
+        - php: '7.2'
+          env: TARGET=phpstan LEVEL=0
+
+        - php: '7.2'
+          env: TARGET=phpstan LEVEL=8
```
Comment (kunicmarko20) [inline .travis.yml]: Maybe allow this one to fail, just for now

File: .travis/script_phpstan.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+
+LEVEL=${1-0}
+
+php vendor/bin/phpstan analyse -l ${LEVEL} src tests
```
Comment (kunicmarko20) [inline .travis/script_phpstan.sh]: New line

File: composer.json [hunk 3]
```diff
@@ -46,14 +46,15 @@
...
-        "twig/twig": "^1.34 || ^2.0"
+        "twig/twig": "^1.34 || ^2.0",
+        "phpstan/phpstan": "^0.9.2"
     },
     "suggest": {
         "egeloen/form-extra-bundle": "Allows to load CKEditor asynchronously"
     },
     "extra": {
         "branch-alias": {
-            "dev-master": "1.x-dev"
```
Comment (kunicmarko20) [inline composer.json]: This is wrong, revert this

File: .travis.yml (hunk 1 above)
Comment (ElectricMaxxx) [reply .travis.yml]: i tried both locally, currently even with level=0 we have 18 issues, shall i fix them here?

Comment (kunicmarko20) [reply .travis.yml]: Yes, we should disable merge if travis is red and that is why I said we should move level 8 to allow failiure.

Comment (ElectricMaxxx) [reply .travis.yml]: so the task is to enable phpstand, run it and fix all issues on level=0?

Comment (kunicmarko20) [reply .travis.yml]: For now, yes. It would be too much to fix everything until level 8 here, so we can do it in a separate PR.

**Comment by ElectricMaxxx** [comment]
@kunicmarko20 any idea why travis isn't triggered anymore? Is caus i also work with gitlab and bitbucket in side project and travis does not like me anymore?

File: .travis.yml [hunk 4]
```diff
@@ -57,7 +63,7 @@ install:
     - if [ -x .travis/install_${TARGET}.sh ]; then .travis/install_${TARGET}.sh; fi;
 
 script:
-    - if [ -x .travis/script_${TARGET}.sh ]; then .travis/script_${TARGET}.sh; fi;
+    - if [ -x .travis/script_${TARGET}.sh ]; then if [ ${TARGET} == "phpstan" ] then .travis/script_${TARGET}.sh ${LEVEL}; else .travis/script_${TARGET}.sh; fi fi;
```
Comment (kunicmarko20) [inline .travis.yml]: maybe missing `;` after first `fi` not really bash expert but this line is problem 100%

**Comment by kunicmarko20** [comment]
Merge https://github.com/FriendsOfSymfony/FOSCKEditorBundle/pull/105 before this PR and your travis will start working.

**Comment by ElectricMaxxx** [comment]
I tried to fix the issues in our templates but phpstan does not accept `@var`  as a variable declartion. Our IDE is happy with it, but phpstan doesn't like it. I also tried to exclude the template files, but for some reason the configuration file is ignored locally, lets wait what travis says.

**Comment by ElectricMaxxx** [comment]
Now it is really green

**Comment by kunicmarko20** [comment]
You can't do it like that, if you check the build you will see that level 7 is nowhere to be seen https://travis-ci.org/FriendsOfSymfony/FOSCKEditorBundle/builds/377813878?utm_source=github_status&utm_medium=notification

**Comment by ElectricMaxxx** [comment]
Ahh. Ok. Thanks.

**Pull request merged**